.github/workflows/interface_tests.yml

```diff
@@ -0,0 +1,27 @@
+name: Interface Tests
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - src/interfaces/coral_web/**
+  pull_request: {}
+
+jobs:
+  interface_tests:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: src/interfaces/coral_web
+
+steps:
+  - name: Checkout repo
+    uses: actions/checkout@v3
+    with:
+      fetch-depth: 2
```

Comment:
leaving this at the default value of `0` fetches the entire history from all branches. Instead, we just fetch the 2 most recent commits to save on time and resources